yjit/src/codegen.rs

```diff
@@ -4021,6 +4021,113 @@ fn jit_rb_false(
     true
 }
 
+
+/// Codegen for Kernel::is_a?
+fn jit_rb_kernel_is_a(
+    jit: &mut JITState,
+    ctx: &mut Context,
+    asm: &mut Assembler,
+    ocb: &mut OutlinedCb,
+    _ci: *const rb_callinfo,
+    _cme: *const rb_callable_method_entry_t,
+    _block: Option<IseqPtr>,
+    argc: i32,
+    known_recv_class: *const VALUE,
+) -> bool {
+    if argc != 1 {
+        return false;
+    }
+
+    // If this is a super class we might not know the class
+    if known_recv_class.is_null() {
+        return false;
+    }
+
+    // Important note: This works because we depend on known_recv_class.
+    // The left hand side here will always have the same type because there
+    // is a guard before this method call that ensures that is the case.
+    // We just guard that the right hand side does not change.
+
+    let sample_rhs = jit_peek_at_stack(jit, ctx, 0);
+    let sample_lhs = jit_peek_at_stack(jit, ctx, 1);
+
+    // If we are testing against a module, we can't be certain
+    // about the class hierarchy.
+    if !unsafe { RB_TYPE_P(sample_rhs, RUBY_T_CLASS) } {
+        return false;
+    }
+    let sample_is_a = unsafe { rb_obj_is_kind_of(sample_lhs, sample_rhs) == Qtrue };
+
+    let side_exit = get_side_exit(jit, ocb, ctx);
+    asm.comment("Kernel::is_a?");
```

Comment:
```suggestion
    asm.comment("Kernel#s_a?");
```
In case you didn't know, `Mod#method` is the common notation for instance methods. `Thing.me` is the notation when `me` is defined on `Thing.singleton_class`.